src/lib.rs

```diff
@@ -571,6 +571,37 @@ impl CecLogicalAddresses {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum TryFromCecLogicalAddressesError {
+    UnknownPrimaryAddress,
+    InvalidPrimaryAddress,
+}
+
+impl TryFrom<cec_logical_addresses> for CecLogicalAddresses {
+    type Error = TryFromCecLogicalAddressesError;
+    fn try_from(addresses: cec_logical_addresses) -> Result<Self, Self::Error> {
+        let primary = CecLogicalAddress::try_from(addresses.primary)
+            .map_err(|_| TryFromCecLogicalAddressesError::UnknownPrimaryAddress)?;
+        let primary = KnownCecLogicalAddress::new(primary)
+            .ok_or(TryFromCecLogicalAddressesError::InvalidPrimaryAddress)?;
```

Comment:
If `KnownCecLogicalAddress` construction fails, should we instead raise `TryFromCecLogicalAddressesError::UnknownPrimaryAddress` error here

Reply:
Fixed